### PR TITLE
feat(resource): Add Export VPN Client Config to awsutils

### PR DESCRIPTION
**Pull request by r351574nc3**
## what
* Adding a feature that is commonly used with TF `provisioner` to export the VPN client config

### Example Use Case
Takes
```hcl
resource "null_resource" "export_client_config" {
  provisioner "local-exec" {
    command = <<-EOT
    mkdir -p ${path.root}/vpn_config/ && \
    aws ec2 export-client-vpn-client-configuration \
            --client-vpn-endpoint-id ${aws_ec2_client_vpn_endpoint.default.id} \
            --output text > ${path.root}/vpn_config/${module.this.id}-client-config-original.ovpn \
            --region ${var.region}
    sed -i ".backup" "s/remote cvpn/remote asdf.cvpn/g" ${path.root}/vpn_config/${module.this.id}-client-config-original.ovpn
    rm ${path.root}/vpn_config/${module.this.id}-client-config-original.ovpn.backup
EOT
  }
```

makes it

```hcl
data "awsutils_ec2_client_vpn_export_client_config" "default" {
    client_vpn_endpoint_id = aws_ec2_client_vpn_endpoint.default.id
}
```

## why
* `provisioner` is a bad smell and isn't easy to maintain since it encapsulates provisioning instructions within TF HCL.
* This is intended to generalize `provisioner` instructions into repeatable/reusable context from a provider.

## references
* N/A

### CODE REVIEW

File: internal/service/ec2/finder/finder.go [hunk 1]
```diff
@@ -5,7 +5,7 @@ import (
 	"github.com/aws/aws-sdk-go/service/ec2"
 )
 
-// VpcDefault looks up the Default Vpc. When not found, returns nil and potentially an API error.
+// InternetGatewayForVPC looks up the Internet Gateway for the given VPC. When not found, returns nil and potentially an API error.
```
Comment (r351574nc3) [inline internal/service/ec2/finder/finder.go:8]: Quick unrelated typo fix in a comment about what the function is doing.

File: examples/provider/provider.tf [hunk 2]
```diff
@@ -15,3 +15,8 @@ provider "awsutils" {
 # Delete default VPC
 resource "awsutils_default_vpc_deletion" "example" {
 }
+
+# Export default VPN client config
+resource "awsutils_export_client_config" "default" {
```
[the remark this answers is not in the document]
Comment (osterman) [reply examples/provider/provider.tf:20]: It seems like the convention by HashiCorp is `aws_ec2_client_vpn....` (E.g. `aws_ec2_client_vpn_endpoint`), so based on that I think it should be `awsutils_ec2_client_vpn_export_client_config`
